app/models/search/bookmark_query.rb

```diff
@@ -72,11 +80,16 @@ def add_owner
   # QUERIES
   ####################
 
-  def parent_child_query
-    [
-      general_query,
-      parent_query
-    ]
+  # Instead of doing a standard query, which would only match bookmark fields
+  # we'll make this a should query that will try to match either the bookmark
+  # or its parent
+  # TODO This isn't right, since it requires all of the fields to be on one or
+  # the other (and, in particular, can't handle tags on the parent with a
+  # specified bookmarker on the child).
```

Comment:
This is AO3-5301/AO3-5320. Rather than make this PR even more complex, I left the bug in.